psp/src/alloc_impl.rs

```diff
@@ -7,6 +7,10 @@ struct SystemAlloc;
 
 unsafe impl GlobalAlloc for SystemAlloc {
     unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        if layout.size() == 0 {
+            return ptr::null_mut();
+        }
+
```

Comment:
This change isn't needed as per `GlobalAlloc::alloc` method's documentation:

> layout must have non-zero size. Attempting to allocate for a zero-sized layout may result in undefined behavior.

https://doc.rust-lang.org/1.83.0/std/alloc/trait.GlobalAlloc.html#tymethod.alloc

Reply:
yeah I know, but was trying to follow this, form the same doc link
> (Extension subtraits might provide more specific bounds on behavior, e.g., guarantee a sentinel address or a null pointer in response to a zero-size allocation request.)

Reply:
> yeah I know, but was trying to follow this, form the same doc link
> 
> > (Extension subtraits might provide more specific bounds on behavior, e.g., guarantee a sentinel address or a null pointer in response to a zero-size allocation request.)

This sentence talks about **extension subtraits**, something like a new `trait GlobalAllocExt: GlobalAlloc`. No need to add this check to the `GlobalAlloc` as no code outside of nested allocators which explicitly depend on `psp::SystemAlloc` would ever depend on such behavior.